tcl/pd-gui.tcl

```diff
@@ -383,9 +383,6 @@ proc init_for_platform {} {
             option add *DialogWindow*font menufont startupFile
             option add *PdWindow*font menufont startupFile
             option add *ErrorDialog*font menufont startupFile
-            # initial dir is home
```

Comment:
The docsdir plugin uses this default to tell when to override the directories at https://github.com/pure-data/pure-data/blob/master/tcl/pd_docsdir.tcl#L126

Is there a default dir when the GUI is started now or will these silly be empty? This is fine, we just might need to double check things.

Maybe a simple fix for the docsdir plugin is to check if "$::filenewdir" is set? If the GUI starts without the file dir paths being set, then the docsdir uses it's path. Otherwise, the panels fall back to home either of the filedir directories do not exist (or are empty) in wheredoesthisgo.tcl, pd_menucommands.tcl, and pdtk_canvas.tcl.

Reply:
there is always a sane default for `filenewdir` and `fileopendir` (i use the two interchangebly)

1. `$::filenewdir` is *always* set to `pwd` in  [pd-gui.tcl:174](https://github.com/pure-data/pure-data/blob/d5b786dc64070229f5b1cd59b0fe2948c0e0c065/tcl/pd-gui.tcl#L174)
1. if Pd is started via the GUI, `$::filenewdir` is always set to `${HOME}` in [pd-gui.tcl:814](https://github.com/pure-data/pure-data/blob/d5b786dc64070229f5b1cd59b0fe2948c0e0c065/tcl/pd-gui.tcl#L814)
1. *Windows* used to be the only platform that would unconditionally `set ::filenewdir $::env(HOME)` afterwards.

*1.* and *2.* establish the sane defaults which i want to see on all platforms.
*Windows* also happens to be the only platform that would ignore the pwd when being started via the Pd-core; this is #478 , and i believe is caused by *3.* and this is what i want my patch attempts to fix.

Reply:
also note that [pd-gui.tcl:174](https://github.com/pure-data/pure-data/blob/d5b786dc64070229f5b1cd59b0fe2948c0e0c065/tcl/pd-gui.tcl#L174) makes any check whether this vars are unset useless (they will always be set).

the only situation that the defaults won't work is if `${HOME}` does not exist (in which case there's a serious problem anyhow). we could nevertheless check for that in [pd-gui.tcl:814](https://github.com/pure-data/pure-data/blob/d5b786dc64070229f5b1cd59b0fe2948c0e0c065/tcl/pd-gui.tcl#L814)

Reply:
Ok. If they are set to HOME before the plugin is loaded, then everything should be fine. The idea is that if the plugin is loaded and they are anything else, the paths should be honored and docspath is not used.

Reply:
>Windows used to be the only platform that would unconditionally set ::filenewdir $::env(HOME) afterwards.

That might be my fault. In any case, I will always applaud consistency of behavior between platforms.